src/main/java/com/jfrog/ide/idea/ui/LocalComponentsTree.java

```diff
@@ -63,22 +63,27 @@ public static LocalComponentsTree getInstance(@NotNull Project project) {
 
     public void addScanResults(List<FileTreeNode> fileTreeNodes) {
         setCellRenderer(new ComponentsTreeCellRenderer());
-        for (FileTreeNode node : fileTreeNodes) {
-            ApplicationManager.getApplication().invokeLater(() -> appendProject(node));
-        }
-    }
-
-    private void appendProject(FileTreeNode filteredRoot) {
-        SortableChildrenTreeNode root;
-        if (getModel() == null) {
-            root = new SortableChildrenTreeNode();
-        } else {
-            root = (SortableChildrenTreeNode) getModel().getRoot();
-        }
+        ApplicationManager.getApplication().invokeLater(() -> {
+            SortableChildrenTreeNode root;
+            if (getModel() == null) {
+                root = new SortableChildrenTreeNode();
+            } else {
+                root = (SortableChildrenTreeNode) getModel().getRoot();
+            }
 
-        root.add(filteredRoot);
-        root.sortChildren();
-        populateTree(root);
+            for (FileTreeNode node : fileTreeNodes) {
+                if (root.getChildren() != null) {
```

Comment:
What do you think about using:
 ```java
Optional.ofNullable(root.getChildren())
                .orElseGet(Collections::emptyList)
```
 instead of the:
```java
if (root.getChildren() != null) {...}
```
?